### ISSUE TITLE
Explore `percent` parameter of DivideDataSet

### ISSUE THREAD

**Issue by DaniBodor** [DeepRank/deeprank-core#183]
This is currently set as a list, but only the first element of the list is read.
If it will always be a 2-way division of the dataset, then set parameter as a float between 0 and 1 (raise error if not in this range).
If we want to allow division into 3, then rewrite the function with a for-loop that reads the number of elements and splits accordingly.

Also, rename parameter because it is not actually a percentage but a fraction.

**Comment by DaniBodor** [comment]
This is currently set as a list, but only the first element of the list is read.
If it will always be a 2-way division of the dataset, then set parameter as a float between 0 and 1 (raise error if not in this range).
If we want to allow division into 3, then rewrite the function with a for-loop that reads the number of elements and splits accordingly.

Also, rename parameter because it is not actually a percentage but a fraction.

**Comment by DaniBodor** [comment]
Yeah, I remembered that there was something about this in another but couldn't find it exactly. 

[train_test_split in sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) uses as default 0.75/0.25 split, so maybe let's just follow their lead? Or does pytorch have a different default?
I do think it's a parameter that you want to open up to the user.

In any case, right now there is not even a check to make sure they add up to 1 or that values above 1 (which would make sense if it would actually be a percentage) fail. 
Haven't looked in detail into your PR; let's wait until that is handled first

**Comment by DaniBodor** [comment]
the `shuffle` parameter in NeuralNet class is regarding a different process than DivideDataSet, right? Shall we rename it in DivideDataSet to avoid confusion?

**Comment by gcroci2** [comment]
> 

Yes, that's right. `shuffle` in NeuralNet is used to shuffle data for the Dataloader, while `shuffle` in DivideDataSet is used to shuffle data before splitting them into train and validation.
Note that DivideDataSet is used only internally and the user does not have access to it (see branch `development`). 

Since DivideDataSet is called only if the user doesn't give a validation set to use during training, I would say that it's fine to fix the percentage at 0.75 and not let the user choose. Same for shufflling. So I would remove both percent and shuffle from inputs of DivideDataSet, and I would set them only inside the function.
Users could always pass in a validation set built by them, and DivideDataSet stays there only for the cases in which the user "forgets" or doesn't know about validation set, and the code split trains in two to validate during training.

**Comment by DaniBodor** [comment]
I can imagine that in some cases a user does not care about how the data is split, but just wants to randomly split it. In this case, I can imagine them wanting to be able to set the fraction of data to use for training and validation. So I would leave the `percent` parameter editable for this case (and see no harm in doing so).

For `shuffle`, I see no reason why you would not want to randomize this, so indeed can go.

**Comment by DaniBodor** [comment]
Let me rephrase: either we shouldn't allow to not give a validation set at all, if we think that one should never want to randomize the data. Or, we should allow to set the split ratio if in some cases a user would want to randomize it.
I don't see why we would allow to not specify which data is which, but not allow to specify the split ratio.

**Comment by gcroci2** [comment]
> I can imagine that in some cases a user does not care about how the data is split, but just wants to randomly split it. In this case, I can imagine them wanting to be able to set the fraction of data to use for training and validation. So I would leave the `percent` parameter editable for this case (and see no harm in doing so).
> 
> For `shuffle`, I see no reason why you would not want to randomize this, so indeed can go.

Okay, I agree that it can be handy to have such an option (even if not necessary because of the flexibility that there is in creating three different sets _before_ calling NeuralNet). Let's keep `percent`. Some conclusive thoughts:
- I would not use a list, but a float/int instead. 
- If float, `percent` indicates the percentage of the train that the user wants to use as training, and validation will be the rest.
- If int, it indicates number of samples that will be randomly selected for train, validation is the rest again.
- `DivideDataSet` can become "_DivideDataSet" because it's used only internally.
